src/parallel/ScatterSplitter.h

```diff
@@ -0,0 +1,93 @@
+
+#include <PTPLib/net/Channel.hpp>
+#include <PTPLib/common/Exception.hpp>
+#include <PTPLib/common/Printer.hpp>
+
+class ScatterSplitter :  public SimpSMTSolver, virtual public Splitter {
+public:
+    ScatterSplitter(SMTConfig & c, THandler & t, PTPLib::net::Channel & ch);
+    PTPLib::net::Channel & getChannel() const   { return channel; }
+    void set_syncedStream(PTPLib::common::synced_stream & ss) { syncedStream = &ss; }  //SMTS Client owns the SyncedStream and should directly set the SyncedStream in ParallelScatterSpitter
+    vec<opensmt::pair<int,int>> const &  get_solver_branch()  const  { return solverBranch; }
+    void set_solver_branch(std::string solver_branch);
+    void mapSolverBranchToFrameId(uint32_t fid, vec<opensmt::pair<int,int>> && solverAddress) {
+        frameId_solverBranch[fid] = std::move(solverAddress);
+    }
+    void addBranchToFrameId(opensmt::span<opensmt::pair<int, int> const> && solver_branch, uint32_t fid);
+
+    inline vec<opensmt::pair<int,int>> const & get_solverBranch(Var v) { return frameId_solverBranch[theory_handler.getTMap().get_FrameId(v)]; }
+
+    void enterSplittingCycle()          { splitContext.enterSplittingCycle(); }
+
+    bool isSplitTypeNone()     const    { return splitContext.isSplitTypeNone(); }
+
+    void resetSplitType()               { splitContext.resetSplitType(); }
+
+    void setSplitTypeScatter()          { splitContext.setSplitTypeScatter(); }
+
+    int getSplitTypeValue() const       { return splitContext.getSplitTypeValue(); }
+
+    int getSearchCounter()  const       { return search_counter; }
+
+private:
+    int search_counter;
+    PTPLib::net::Channel &  channel;
+    int                     trail_sent;
+    bool                    firstPropagation;
+    int                     numTriviallyPropagatedOnDl0;
+
+    vec<opensmt::pair<int,int>> solverBranch;
+    PTPLib::common::synced_stream * syncedStream;
+    using map_frameId_solverBranch = std::map<uint32_t, vec<opensmt::pair<int,int>>>;
+    map_frameId_solverBranch frameId_solverBranch;
+    using map_var_frameId = std::map<Var ,uint32_t>;
+    map_var_frameId var_frameId;
+
+    void runPeriodic() override;                                       // Check if solver is in clause share mode to starts clause exposing operation
+
+protected:
+    std::vector<vec<Lit>> split_assumptions;
+    bool     scatterLevel();                                                  // Are we currently on a scatter level.
+    opensmt::pair<SplitData,lbool> createSplitAndBlockAssumptions();          // Create a split formula and place it to the splits vector.
+    bool     excludeAssumptions(vec<Lit> const & neg_constrs);                // Add a clause to the database and propagate
+    inline bool isAssumptionVar(Var  v)   const &   { return theory_handler.getTMap().isAssumptionVar(v); }
+
+    lbool solve_() override;
+    bool branchLitRandom() override;
+    Var doActivityDecision() override;
+    bool okContinue() override;
```

Comment:
One option would be to place it as the first thing in the corresponding `while` loop.  We'd need to have it directly in `CoreSMTSolver`, but I don't think this would be a bad idea.  Maybe this is minor, but I'd prefer the function as a `const`.